src/hostapi/alsa/pa_linux_alsa.c

```diff
@@ -1450,7 +1456,7 @@ static PaError BuildDeviceList( PaAlsaHostApiRepresentation *alsaApi )
 
         PA_ENSURE( FillInDevInfo( alsaApi, hwInfo, blocking, devInfo, &devIdx ) );
     }
-    assert( devIdx < numDeviceNames );
+    assert( devIdx <= numDeviceNames );
```

Comment:
What is the reason for this change? It is not mentioned in the commit message.

Reply:
It is mentioned `Fixed logic of guarding assertion in BuildDeviceList() when only one PCM device is available (may occur if PA_ALSA_IGNORE_ALL_PLUGINS is defined).`. 

When `PA_ALSA_IGNORE_ALL_PLUGINS` is defined, e.g. user compiled PA to ignore all ALSA plugins, then `assert( devIdx < numDeviceNames )` will become true and process will stop.

Reply:
The algorithm of the failing assert is the following:

Assume device has only USB DAC attached as playback device.

1. `numDeviceNames` becomes 1
2. `FillInDevInfo` (line 1457) increments `devIdx` to 1
3. loop at line 1448 `for( i = 0, devIdx = 0; i < numDeviceNames; ++i )` exits as there is only one PCM device (`numDeviceNames == 1`)
4. assert condition becomes false `assert(1 < 1)`